Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/Adaptive_sizing_field.h

```diff
@@ -0,0 +1,337 @@
+      }
+      else if (vertex_size_sq < CGAL::square(m_short))
+      {
+        put(m_vertex_sizing_map, v, m_short);
+#ifdef CGAL_PMP_REMESHING_VERBOSE
+        ++undersize;
+#endif
+      }
+      else
+      {
+        put(m_vertex_sizing_map, v, CGAL::approximate_sqrt(vertex_size_sq));
+#ifdef CGAL_PMP_REMESHING_VERBOSE
+        ++insize;
+#endif
+      }
+    }
+#ifdef CGAL_PMP_REMESHING_VERBOSE
+    std::cout << " done (" << insize << " from curvature, "
+              << oversize  << " set to max, "
+              << undersize << " set to min)" << std::endl;
+#endif
+  }
+
+  FT sqlength(const vertex_descriptor va,
+              const vertex_descriptor vb) const
+  {
+    return FT(CGAL::squared_distance(get(m_vpmap, va), get(m_vpmap, vb)));
+  }
+
+  FT sqlength(const halfedge_descriptor& h, const PolygonMesh& pmesh) const
+  {
+    return sqlength(target(h, pmesh), source(h, pmesh));
+  }
+
+public:
+  FT get_sizing(const vertex_descriptor v) const
+  {
+    CGAL_assertion(get(m_vertex_sizing_map, v));
+    return get(m_vertex_sizing_map, v);
+  }
+
+  std::optional<FT> is_too_long(const halfedge_descriptor h, const PolygonMesh& pmesh) const
+  {
+    const FT sqlen = sqlength(h, pmesh);
+    FT sqtarg_len = CGAL::square(4./3. * (CGAL::min)(get(m_vertex_sizing_map, source(h, pmesh)),
+                                                     get(m_vertex_sizing_map, target(h, pmesh))));
+    CGAL_assertion(get(m_vertex_sizing_map, source(h, pmesh)));
+    CGAL_assertion(get(m_vertex_sizing_map, target(h, pmesh)));
+    if(sqlen > sqtarg_len)
+      return sqlen;
+    else
+      return std::nullopt;
+  }
+
+  std::optional<FT> is_too_long(const vertex_descriptor va, const vertex_descriptor vb) const
+  {
+    const FT sqlen = sqlength(va, vb);
+    FT sqtarg_len = CGAL::square(4./3. * (CGAL::min)(get(m_vertex_sizing_map, va),
+                                                     get(m_vertex_sizing_map, vb)));
+    CGAL_assertion(get(m_vertex_sizing_map, va));
+    CGAL_assertion(get(m_vertex_sizing_map, vb));
+    if (sqlen > sqtarg_len)
+      return sqlen;
+    else
+      return std::nullopt;
+  }
+
+  std::optional<FT> is_too_short(const halfedge_descriptor h, const PolygonMesh& pmesh) const
+  {
+    const FT sqlen = sqlength(h, pmesh);
+    FT sqtarg_len = CGAL::square(4./5. * (CGAL::min)(get(m_vertex_sizing_map, source(h, pmesh)),
+                                                     get(m_vertex_sizing_map, target(h, pmesh))));
+    CGAL_assertion(get(m_vertex_sizing_map, source(h, pmesh)));
+    CGAL_assertion(get(m_vertex_sizing_map, target(h, pmesh)));
+    if (sqlen < sqtarg_len)
+      return sqlen;
+    else
+      return std::nullopt;
+  }
+
+  Point_3 split_placement(const halfedge_descriptor h, const PolygonMesh& pmesh) const
+  {
+    return midpoint(get(m_vpmap, target(h, pmesh)),
+                    get(m_vpmap, source(h, pmesh)));
+  }
+
+  void update_sizing_map(const vertex_descriptor v, const PolygonMesh& pmesh)
```

Comment:
Currently I need it public because of

https://github.com/janetournois/cgal/blob/041804c2694944ce94d69b96fbec62dd13f17230/Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/internal/Isotropic_remeshing/remesh_impl.h#L435-L436

and

https://github.com/janetournois/cgal/blob/041804c2694944ce94d69b96fbec62dd13f17230/Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/internal/Isotropic_remeshing/remesh_impl.h#L551-L552

Add it to the concept maybe?

Reply:
indeed...
Then I would suggest to rename the function too `sizing.update(v, pmesh)` to be more generic, and add it to the concept.
Then, you can remove the calls to `std::is_same_v`

@sloriot do you agree with that?

Reply:
Should I just return the empty function for the uniform field in case I remove `std::is_same_v`?

Reply:
Yes!

Reply:
Done in 298d9e9c944.